### ISSUE TITLE
Compiler error when trying to add a function

### ISSUE THREAD

**Issue by dedesite** [vlang/v#14371]
**V version:** V 0.2.4 cd4fa04
**OS:** Ubuntu 20.04

**What did you do?**

I try to add a function in my code base (a v clone of chibicc), sorry didn't manage to isolate the bug

You can find a copy of the file here : https://gist.github.com/dedesite/024eac54e5caf10dfbfbfdf0f2bf52e3

**What did you expect to see?**

The compiler to warn me if there is an error in my new function or to compile successfully
 
**What did you see instead?**
It was just after adding the function `unary`. 
I tried :
* to name it otherwise -> same error
* to move it below other function -> same error
* to change it's return types -> got a proper compiler error
* to remove it's content -> got a proper compiler error
* to just let the `primary` call at the end -> same error

```
v main.v -o chibivcc
V panic: array.get: index out of range (i == 1, a.len == 1)
v hash: cd4fa04
/tmp/v_1000/v2.3142598272497904992.tmp.c:17423: at _v_panic: Backtrace
/tmp/v_1000/v2.3142598272497904992.tmp.c:16939: by array_get
/tmp/v_1000/v2.3142598272497904992.tmp.c:59614: by v__checker__Checker_return_stmt
/tmp/v_1000/v2.3142598272497904992.tmp.c:51375: by v__checker__Checker_stmt
/tmp/v_1000/v2.3142598272497904992.tmp.c:51809: by v__checker__Checker_stmts_ending_with_expression
/tmp/v_1000/v2.3142598272497904992.tmp.c:51785: by v__checker__Checker_stmts
/tmp/v_1000/v2.3142598272497904992.tmp.c:58341: by v__checker__Checker_if_expr
/tmp/v_1000/v2.3142598272497904992.tmp.c:52151: by v__checker__Checker_expr
/tmp/v_1000/v2.3142598272497904992.tmp.c:51300: by v__checker__Checker_stmt
/tmp/v_1000/v2.3142598272497904992.tmp.c:51809: by v__checker__Checker_stmts_ending_with_expression
/tmp/v_1000/v2.3142598272497904992.tmp.c:51785: by v__checker__Checker_stmts
/tmp/v_1000/v2.3142598272497904992.tmp.c:56359: by v__checker__Checker_fn_decl
/tmp/v_1000/v2.3142598272497904992.tmp.c:51332: by v__checker__Checker_stmt
/tmp/v_1000/v2.3142598272497904992.tmp.c:49194: by v__checker__Checker_check
/tmp/v_1000/v2.3142598272497904992.tmp.c:49259: by v__checker__Checker_check_files
/tmp/v_1000/v2.3142598272497904992.tmp.c:27397: by v__builder__Builder_middle_stages
/tmp/v_1000/v2.3142598272497904992.tmp.c:27428: by v__builder__Builder_front_and_middle_stages
/tmp/v_1000/v2.3142598272497904992.tmp.c:29866: by v__builder__cbuilder__gen_c
/tmp/v_1000/v2.3142598272497904992.tmp.c:29889: by v__builder__cbuilder__build_c
/tmp/v_1000/v2.3142598272497904992.tmp.c:29861: by v__builder__cbuilder__compile_c
/tmp/v_1000/v2.3142598272497904992.tmp.c:28845: by v__builder__compile
/tmp/v_1000/v2.3142598272497904992.tmp.c:30017: by main__main
/tmp/v_1000/v2.3142598272497904992.tmp.c:30910: by main
make: *** [Makefile:5: chibivcc] Error 1
```

**Comment by danieldaeschle** [comment]
small reproduceable:
```
fn test() ?(&int, &int) {
	return test() ?
}

fn main() {

}
```

**Comment by dedesite** [comment]
Nice catch !

**Comment by spytheman** [comment]
A workaround is `return unary(tok.next) ?` -> `return unary(tok.next)` . 

The panic is in the branch that checks for that error.

I will push a fix to the compiler soon, after the tests pass locally.

**Comment by spytheman** [comment]
The fix is in b7ca4c1, thanks to the small reproduction that @danieldaeschle made, it was easy to correct.
